safekeeper/src/wal_storage.rs

```diff
@@ -604,8 +604,7 @@ impl WalReader {
 
         // Try to open remote file, if remote reads are enabled
         if self.enable_remote_read {
-            let (reader, _) = read_object(wal_file_path, xlogoff as u64).await;
-            return Ok(Box::pin(reader));
+            return read_object(wal_file_path, xlogoff as u64).await;
```

Comment:
Will it read straight from s3 without saving it locally? So if we have a replica that constantly connects and craches after reading a few bytes it will every time connect to s3? Also if we have more than one replica, will they share the file or every replica will get its own download stream from s3?

Reply:
Not introduced with this PR though, so probably question to @petuhovskiy or @arssher

Reply:
Yes, it reads directly from s3 and doesn't save to disk which allows to not think about garbage collecting files. This proxying to s3 is sort of emergency code path; replica shouldn't regularly crash after receiving some bytes. So while it can be improved, currently it's ok in IMV.